### ISSUE TITLE
Windows menu (menubar) is not populated

### ISSUE THREAD

**Issue by iandobbie** [MicronOxford/cockpit#683]
On windows 10 the windows menu isn't properly populated. It has reset window positons, Launch ValueLogViewer and a dsp debug window but none of the other windows are listed. This means it is hard to get back a lost or buried
![Screenshot 2020-10-22 at 11 19 43](https://user-images.githubusercontent.com/5451909/96858715-8cfcfe00-1458-11eb-9792-731ab4a65d99.png)
 window, especially on a small screen.

**Comment by carandraug** [comment]
I can't reproduce the issue with a Windows 10 VM and I've seen others using it on a real Windows 10. Are you using MicronOxford's master branch or some other master branch? Can you investigate what's happening on that system (see `cockpit.gui.mainWindow.WindowsMenu.OnMenuOpen`)?

**Comment by iandobbie** [comment]
The branch is micronoxford/master as of this morning. I will investigate what happening on that call.

**Comment by iandobbie** [comment]
Installed pycharm and played around a bit. As far as I can tell the OnMenuOpen is not getting called, hence the issue about it not populating the menu.

**Comment by carandraug** [comment]
Check with the event watcher what is happening to it (it's pretty unreliable but may give some hints):

```
from wx.lib.inspection import InspectionTool
InspectionTool().Show() # then click on the big Events button in the tool bar
```

**Comment by iandobbie** [comment]
From the wx documentation 

EVT_MENU_OPEN(func):
A menu is about to be opened. On Windows, this is only sent once for each navigation of the menubar (up until all menus have closed).

I don't understand why mousing over the menu and selecting it doesn't run this function but it isn't being called in pycharm.

**Comment by iandobbie** [comment]
I wonder if the menu_open event only happens on the parent, I guess the menubar.

**Comment by carandraug** [comment]
I got my hands on the machine that had this problem, and the problem seems to have been the wxPython version (it was running wxPython 3.0.6 with wxWidgets version 3.0.5).  I installed wxPython 4.1.0 and the problem was solved. Maybe we should just require wxPython 4.1.

**Comment by iandobbie** [comment]
Well spotted. There is a wx issue on Mac that is still unsolved. We should make sure we don’t break the Mac version while fixing the pc one.

Ian

Sent from my iPhone

On 24 Oct 2020, at 02:02, Carnë Draug <notifications@github.com> wrote:

﻿

I got my hands on the machine that had this problem, and the problem seems to have been the wxPython version (it was running wxPython 3.0.6 with wxWidgets version 3.0.5). I installed wxPython 4.1.0 and the problem was solved. Maybe we should just require wxPython 4.1.
